PostHog/PostHogQueue.swift

```diff
@@ -157,36 +172,22 @@ class PostHogQueue {
                 self.isFlushing = true
             }
 
-            let datas = self.fileQueue.peek(count)
+            let items = self.fileQueue.peek(count)
 
             var processing = [PostHogEvent]()
-            var deleteIndexes = [Int]()
-            for (index, data) in datas.enumerated() {
+
+            for item in items {
                 // each element is a PostHogEvent if fromJSON succeeds
-                guard let event = PostHogEvent.fromJSON(data) else {
-                    deleteIndexes.append(index)
+                guard let event = PostHogEvent.fromJSON(item) else {
                     continue
                 }
                 processing.append(event)
             }
 
-            // delete files that aren't valid as a event JSON
-            if !deleteIndexes.isEmpty {
-                for index in deleteIndexes {
-                    // TOOD: check if the indexes dont move and delete the wrong files
-                    // might need an array of file paths instead of the data, so we know what to delete correctly
-                    self.fileQueue.delete(index: index)
-                }
-            }
-
-            if processing.isEmpty {
-                return
-            }
-
             completion(PostHogConsumerPayload(events: processing) { success in
                 hedgeLog("Completed!")
                 if success {
-                    _ = self.fileQueue.pop(processing.count)
+                    self.fileQueue.pop(items.count)
```

Comment:
We pop the total items from the peek method, even if some of them failed to be parsed, so we don't run into the problem with the deletion, the index would be moving and its just harder.